src/test/test_failure_slice.cpp

```
@@ -0,0 +1,284 @@
+    const std::string sample = "ebpf-samples/build/exposeptr.o";
+    if (!sample_exists(sample)) {
+        SKIP("Sample file not found: " << sample);
+    }
+    auto slices = get_failure_slices(sample, ".text");
+
+    REQUIRE(slices.size() == 1);
+    REQUIRE(slices[0].relevance.size() > 0);
+
+    // The failure involves map update with registers r1, r3
+    auto& failing_relevance = slices[0].relevance.at(slices[0].failing_label);
+    // At minimum r3 should be relevant (the value being stored)
```

Cppcheck (2.19.0)

[style] The function 'udiv' is never used.

(unusedFunction)